tests/crypto/README.rst

```diff
@@ -0,0 +1,342 @@
+* :file:`overlay-backend-oberon.conf` - uses the Oberon software library only for all cryptographic operations
+
+.. _crypto_test_ztest_custom:
+
+Ztest custom log formatting
+===========================
+
+Cryptography tests replace the standard Ztest formatting to assure more efficient reporting of running tests and test results.
+Set the configuration option :option:`CONFIG_ZTEST_TC_UTIL_USER_OVERRIDE` to replace the Ztest macros ``TC_START`` and ``Z_TC_END_RESULT`` with versions more suited for reporting results of cryptographic tests.
+
+:option:`CONFIG_ZTEST_TC_UTIL_USER_OVERRIDE` uses :file:`tests/crypto/include_override/tc_util_user_override.h` to define the custom formatting.
+
+.. _crypto_test_testing:
+
+Testing
+=======
+
+1. Compile and program the application.
+#. Observe the result of the different test vectors in the log using `RTT Viewer`_ or a terminal emulator.
+   The last line of the output indicates the test result::
+
+      PROJECT EXECUTION SUCCESSFUL
+
+
+Additional test cases and test vectors
+======================================
+
+Test cases and test vectors can be added to the test suite either by including additional source files or by extending the existing files.
+
+
+Test case
+---------
+
+A test case is a function designed to verify parts of the functionality of a cryptographic operation.
+Most cryptographic operations, like hash calculations and ECDH, have multiple test cases to be able to cover all features.
+A typical test case is called by looping over the registered test vectors and calling the test case.
+The execution logs the verdict for each test vector.
+
+Registering a test case
+~~~~~~~~~~~~~~~~~~~~~~~
+
+A new test case must be registered to the ``test_case_data`` section using ITEM_REGISTER, which registers it with the named section ``test_case_hmac_data``::
+
+   ITEM_REGISTER(test_case_hmac_data, test_case_t test_hmac) = {
+   .p_test_case_name = "HMAC",
+   .setup = hmac_setup,
+   .exec = exec_test_case_hmac,
+   .teardown = hmac_teardown,
+   .vector_type = TV_HMAC,
+   .vectors_start = __start_test_vector_hmac_data,
+   .vectors_stop = __stop_test_vector_hmac_data,
+   };
+
+
+.. note::
+   The macro call to ITEM_REGISTER must be done in a .c file.
+
+Setting up a test case
+~~~~~~~~~~~~~~~~~~~~~~
+
+As part of the test case setup, any previously used buffers are cleared.
+Then, the next test vector is fetched using the ``ITEM_GET`` macro.
+The macro requires the following parameters:
+ * ``test_vector_hmac_data`` - the section to fetch the test vector from (HMAC in this example).
+ * ``test_vector_hmac_t`` - information about which type of test vector to expect in the given section.
+   In the example, ``test_vector_hmac_t`` is expected, the same type that is used when registering HMAC test vectors.
+ * Information about which index to fetch a test vector from.
+
```

Comment:
The last setup step is not described, maybe add something like:
```suggestion
Lastly, the test vector fetched is unhexified.
Test vector data is stored as strings of hexadecimal characters.
In order to use them, they must be parsed to binary, which this step does.
```

Reply:
added in a slightly changed form , please checke @torsteingrindvik

Reply:
> The fetched test vector is then unhexified and its data is stored as strings of hexadecimal characters.

Could you reword this? It sounds like the data is actively at runtime being stored as hex chars by unhexify.

Reply:
split the sentence into two